jest/__mocks__/html-rewriter.js

```diff
@@ -1,8 +1,36 @@
 export const mockHtmlRewritingStream = jest.fn();
-export const mockOnElement = jest.fn();
+export const mockOnElement = jest.fn().mockImplementation(() => {
+  return {
+    selector: String,
+    handler: onElement()
+  };
+});
 export const mockReadableStream = jest.fn().mockReturnThis();
 export const mockWritableStream = jest.fn().mockReturnThis();
 export const mockWritableStreamPipeThrough = jest.fn().mockReturnThis();
+export const mockAfter = jest.fn();
+export const mockAppend = jest.fn();
+export const mockBefore = jest.fn();
+export const mockGetAttribute = jest.fn();
+export const mockPrepend = jest.fn();
+export const mockRemoveAttribute = jest.fn();
+export const mockReplaceChildren = jest.fn();
+export const mockReplaceWith = jest.fn();
+export const mockSetAttribute = jest.fn();
+
+export const onElement = jest.fn().mockImplementation(() => {
+  return {
+    after: mockAfter,
+    append: mockAppend,
+    before: mockBefore,
+    getAttribute: mockGetAttribute,
+    prepend: mockPrepend,
+    removeAttribute: mockRemoveAttribute,
+    replaceChildren: mockReplaceChildren,
+    replaceWith: mockReplaceWith,
+    setAttribute: mockSetAttribute
+  };
+});
```

Comment:
It would be worth writing a test that uses this. ie, verify that a tag handler behaves as expected.